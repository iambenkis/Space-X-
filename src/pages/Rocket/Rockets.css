.rockets {
    display: flex;
    flex-direction: column;
    width: 100%;
    font-size: 1.7rem;
  }

  .rocket {
    display: flex;
    margin: 2rem 10%;
  }

  .rocket .rocket-text {
    margin-top: 1rem;
    margin-left: 2rem;
  }

  .rocket-text h2 {
    margin-bottom: 1rem;
  }

  .rocket-text p {
    font-size: 1.6rem;
    line-height: 1.6;
    margin-bottom: 1rem;
  }

  .rocket-text button {
    color: #fff;
    border: none;
    padding: 1rem 1.7rem;
    border-radius: 5px;
    cursor: pointer;
  }

  .rocket-text .reserve{
    background-color: #0290ff;
  }

  .rocket-text .cancel{
    background-color: #7c7c7c;
  }

  .rocket-text button:hover {
    background-color: #0472c7;
  }

  .rocket-text .cancel:hover {
    background-color: #c9c9c9;
  }
  .rocket img {
    width: 27rem;
    height: 20rem;
    border: 1px solid rgb(173, 173, 173);
    padding: .3rem;
    border-radius: 7px;
  }

  .rocket span {
    background-color: #0290ff;
    border-radius: 5px;
    margin-right: .5rem;
  }
