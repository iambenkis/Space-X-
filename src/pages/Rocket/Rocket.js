import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getRockets } from '../../redux/Rockets/Rockets/rockets';

const Rocket = () => {
  const rockets = useSelector(state => state.rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getRockets());
  }, []);
  console.log(rockets)
return (
  <>
    <p>Rocket Page</p>
  </>
);
}
export default Rocket;
