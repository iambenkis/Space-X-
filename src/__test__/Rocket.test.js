import React from "react";
import '@testing-library/jest-dom';
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import store from "../redux/Rockets/configurationStore";
import Rocket from '../pages/Rocket/Rocket'

describe("Test Rocket component ", () => {
    test('Does Rockets match the snapshot ?', () => {
        const rocket = renderer.create(
             <BrowserRouter>
                 <Provider store={store}>
                    <Rocket />
                 </Provider>
             </BrowserRouter>
         )
         expect(rocket).toMatchSnapshot();
     })
})
