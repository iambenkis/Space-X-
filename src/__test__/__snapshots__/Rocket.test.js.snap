// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocket component  Does Rockets match the snapshot ? 1`] = `
<div
  className="rockets"
/>
`;
